networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -588,6 +595,19 @@ def _get_snat_pool_name(self, subnet):
         snat_prefix = self._snat_prefix(subnet)
         return "%s_nat_pool" % (snat_prefix)
 
+    def _add_default_route(self, ri, ext_gw_port):
+        if self._fullsync and (ri.router_id in
+                               self._existing_cfg_dict['routes']):
+            LOG.debug("Default route already exists, skipping")
+            return
+        ext_gw_ip = ext_gw_port['hosting_info']['gateway_ip']
+        if ext_gw_ip:
+            vrf_name = self._get_vrf_name(ri)
+            out_itfc = self._get_interface_name_from_hosting_port(ext_gw_port)
+            conf_str = asr1k_snippets.SET_DEFAULT_ROUTE_WITH_INTF % (
+                vrf_name, out_itfc, ext_gw_ip)
+            self._edit_running_config(conf_str, 'SET_DEFAULT_ROUTE_WITH_INTF')
```

Comment:
This is interesting -- this code was leveraged from the stock ASR1k driver. However, I can't find an actual member called _existing_cfg_dict. However, there is a member of the same name without the leading underscore (i.e. existing_cfg_dict) in asr1k_cfg_syncer. I'm guessing that there's no exception because this code path is never taken under _fullsync. 

That said, I'm think the answer is no -- this is probably intended to compare against what the config syncer has when doing an intial clean, and therefore isn't needed afterwards (i.e. a new full-sync would cause the cfg syncer to generate a new existing_cfg_dict).

I'll look into removing the leading underscore though.